airflow/models/__init__.py

```diff
@@ -1021,9 +1027,8 @@ def are_dependents_done(self, session=None):
         count = ti[0][0]
         return count == len(task.downstream_task_ids)
 
-    @property
     @provide_session
-    def previous_ti(self, session=None):
+    def get_previous_ti(self, session=None):
```

Comment:
Actually, since the property is the "preferred" way could we name the method `_get_previous_ti` please? (otherwise this might show up in generated API docs and cause confusion as which one is the "recommended" one to call)

Or do you think it's okay to have both?

Reply:
I agree, revised to make the new methods private and put docstrings in the public properties.